### PR TITLE
make winapi a windows-specific dependency

### LINKED ISSUE #362: uuid now depends on winapi on all platforms
This is unfortunate and we (Fedora) would appreciate if you would mark it as needed only on Windows.

Thanks!

### PR DESCRIPTION
**Pull request by kinggoesgaming**
**I'm submitting a(n)** bug fix

# Description
Using `[target.'cfg()'` format to only link to winapi only on windows

# Motivation
Hey fedora doods, you happy now?

# Tests
tests passing

# Related Issue(s)
closes #362

### CODE REVIEW

**Comment by kinggoesgaming** [comment]
hahahaha. will fix this soon

File: Cargo.toml [hunk 1]
```diff
@@ -89,3 +85,7 @@ u128 = ["byteorder"]
 const_fn = ["nightly"]
 # Nightly marker feature gate
 nightly = []
+
+[target.'cfg(all(windows, target_env="msrv"))'.dependencies.winapi]
```
Comment (KodrAus) [inline Cargo.toml]: Should this be `msvc`?

Comment (KodrAus) [reply Cargo.toml]: Also, do we even need the `target_env` here? If you're building with a GNU toolchain on Windows you might still want `winapi`, right?

Comment (kinggoesgaming) [reply Cargo.toml]: > Should this be `msvc`?

oops



> Also, do we even need the `target_env` here? If you're building with a GNU toolchain on Windows you might still want `winapi`, right?

true. will adjust this when I can

**Comment by kinggoesgaming** [comment]
@uuid-rs/uuid this should be good to go when the build checks are done

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by kinggoesgaming** [comment]
bors r=@Dylan-DPC

**Comment by ignatenkobrain** [comment]
Thanks!

**Pull request merged**